compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7779,7 +7784,7 @@ private BType getAllFieldType(BRecordType recordType) {
 
     private boolean checkValidJsonOrMapLiteralKeyExpr(BLangExpression keyExpr, boolean computedKey, AnalyzerData data) {
         if (computedKey) {
-            checkExpr(keyExpr, symTable.stringType, data);
+            exprIncompatible(symTable.stringType, keyExpr, data);
```

Comment:
codecov warning?

Reply:
Fixed after rebase

Reply:
Is the subsequent logic here (L7901 - L7904) correct? 

I'm not sure if this along with the changes in L7769 - L7773 are correct. Ideally, we should have the same flow, just that on one path we have non-error logging checks.